tools/errorfinder/error_finder.go

```diff
@@ -0,0 +1,49 @@
+package errorfinder
```

Comment:
Can we have a one sentence summary about the _purpose_ of the code in this file? It's not clear initially for me just by looking at the exported interface and constructor function. (What does "finding" means, why we are doing it, what's the output of finding the error)